saleor/static/dashboard-next/discounts/components/VoucherDetailsPage/VoucherDetailsPage.tsx

```diff
@@ -0,0 +1,252 @@
+import * as React from "react";
+
+import CardSpacer from "../../../components/CardSpacer";
+import { ConfirmButtonTransitionState } from "../../../components/ConfirmButton";
+import Container from "../../../components/Container";
+import Form from "../../../components/Form";
+import Grid from "../../../components/Grid";
+import PageHeader from "../../../components/PageHeader";
+import SaveButtonBar from "../../../components/SaveButtonBar";
+import { Tab } from "../../../components/Tab";
+import TabContainer from "../../../components/Tab/TabContainer";
+import i18n from "../../../i18n";
+import { maybe } from "../../../misc";
+import { ListProps, UserError } from "../../../types";
+import {
+  VoucherDiscountValueType,
+  VoucherType
+} from "../../../types/globalTypes";
+import { VoucherDetails_voucher } from "../../types/VoucherDetails";
+import DiscountCategories from "../DiscountCategories";
+import DiscountCollections from "../DiscountCollections";
+import DiscountProducts from "../DiscountProducts";
+import VoucherCountries from "../VoucherCountries";
+import VoucherInfo from "../VoucherInfo";
+import VoucherOptions from "../VoucherOptions";
+import VoucherSummary from "../VoucherSummary";
+
+export enum VoucherDetailsPageTab {
+  categories = "categories",
+  collections = "collections",
+  products = "products"
+}
+export function voucherDetailsPageTab(tab: string): VoucherDetailsPageTab {
+  return tab === VoucherDetailsPageTab.products
+    ? VoucherDetailsPageTab.products
+    : tab === VoucherDetailsPageTab.collections
+    ? VoucherDetailsPageTab.collections
+    : VoucherDetailsPageTab.categories;
+}
+
+export interface FormData {
+  applyOncePerOrder: boolean;
+  code: string;
+  discountType: VoucherDiscountValueType;
+  endDate: string;
+  minAmountSpent: number;
+  name: string;
+  startDate: string;
+  type: VoucherType;
+  usageLimit: number;
+  value: number;
+}
+
+export interface VoucherDetailsPageProps
+  extends Pick<ListProps, Exclude<keyof ListProps, "onRowClick">> {
+  activeTab: VoucherDetailsPageTab;
+  defaultCurrency: string;
+  errors: UserError[];
+  saveButtonBarState: ConfirmButtonTransitionState;
+  voucher: VoucherDetails_voucher;
+  onBack: () => void;
+  onCategoryAssign: () => void;
+  onCategoryUnassign: (id: string) => void;
+  onCategoryClick: (id: string) => () => void;
+  onCollectionAssign: () => void;
+  onCollectionUnassign: (id: string) => void;
+  onCollectionClick: (id: string) => () => void;
+  onCountryAssign: () => void;
+  onCountryUnassign: (code: string) => void;
+  onProductAssign: () => void;
+  onProductUnassign: (id: string) => void;
+  onProductClick: (id: string) => () => void;
+  onRemove: () => void;
+  onSubmit: (data: FormData) => void;
+  onTabClick: (index: VoucherDetailsPageTab) => void;
```

Comment:
I think U could extract a base interface and reuse most of this with `SaleDetailsPageProps` and mb others.

Reply:
I could, but I'm not convinced if I should - again, if two things look similar, it doesn't mean that they, in fact, are similar.

Reply:
IMO if they share the same abstraction I don't see point not doing so.